src/PhysicsEngine.cpp

```diff
@@ -179,6 +180,273 @@ void ForwardEuler::march_step(double tStamp, double tStep) {
+    /*************** Retrieve Instantaneous Rocket Parameters *****************/
+
+    Vector3d Cp_vect_rf = rocket_.get_Cp_vect();
+
+    double inertia[9];  // moments of inertia
+    rocket_.get_I(inertia);
+
+    double area = rocket_.get_A_ref();
+    double c_Na = rocket_.get_Cna();  // normal force coefficient derivative
+    double drag_coef = rocket_.get_Cd();
+
+    /************************ Calculate Net Torque ***************************/
+
+    Vector3d aero_force_rf;
+    Vector3d aero_torque_rf;
+    Vector3d aero_force_if;
+    Vector3d aero_torque_if;
+    Vector3d net_force_rf;
+    Vector3d net_torque_rf;
+
+    if (vel_if.norm() > 0.01) {
+        Vector3d vel_rf = rocket_.i2r(vel_if);
+        Vector3d normal_force_rf;
+
+        // angle between velocity vector and rocket axis
+        double alpha = acos(vel_rf.z() / vel_rf.norm());
+
+        double normal_coef = c_Na * alpha;
+
+        double normal_force_mag = 0.5 * normal_coef * vel_rf.squaredNorm() *
+                                  area * Atmosphere::get_density(pos_if.z());
+        normal_force_rf = {(-vel_rf.x()), (-vel_rf.y()), 0};
+
+        normal_force_rf.normalize();
+        normal_force_rf = normal_force_rf * normal_force_mag;
+
+        double drag_mag = 0.5 * drag_coef * vel_rf.squaredNorm() * area *
+                          Atmosphere::get_density(pos_if.z());
+        Vector3d drag_rf{0, 0, std::copysign(drag_mag, -vel_rf.z())};
+
+        aero_force_rf = normal_force_rf + drag_rf;
+        aero_torque_rf = Cp_vect_rf.cross(aero_force_rf);
+    } else {
+        aero_torque_rf = {0, 0, 0};
+    }
+
+    Vector3d net_torque_if = rocket_.r2i(aero_torque_rf);
+
+    return net_torque_if;
+}
+
+/**
+ * @brief Calculates a possible rocket state based on the initial and inputed
+ * state
+ *
+ * Method performs a basic Euler step on the velocity and angular velocity of
+ * the rocket, as well as recalculating forces, accereration, angular
+ * acceleration, and orientation, using the initial state of the rocket and the
+ * data from the state an inputed state
+ *
+ * @param k The state of the rocket being used to calculate the next state
+ * @return RungeKuttaState  Velocity, accereration, angular velocity, and
+ * angular acceleration at a moment
+ */
+RungeKutta::RungeKuttaState RungeKutta::calc_state(double tStamp, double tStep,
+                                                   RungeKuttaState k) {
+    // the rocket's initial state (k1), regardless of which state is being
+    // calculated
+    Vector3d pos_initial = rocket_.get_r_vect();
+    Vector3d vel_initial = rocket_.get_r_dot();
+    Vector3d ang_vel_initial = rocket_.get_w_vect();
+    Quaterniond orient_true = rocket_.get_q_ornt();
+
+    double inertia[9];
+    rocket_.get_I(inertia);
+
+    Quaterniond orient = update_quaternion(orient_true, k.ang_vel, tStep);
+    rocket_.set_q_ornt(orient);  // sets the orientation to the current state in
```

Comment:
I'm still not a fan of writing intermediate values into the Rocket class and then resetting it back to default. Feels like bad software design IMO.

I see that we only have to do this because the `calc_net_*` functions use the `r2i/i2r` functions of the rocket class. I think in the future we should have `r2i/i2r` helper functions in `PhysicsEngine` as well that take take in whatever quaternion you want to pass in.

I don't think this blocks merging the PR, but if you agree, then please make a ticket to implement this later.